tests/unit_tests/dft/source/descriptor_tests.cpp

```diff
@@ -0,0 +1,453 @@
+*  limitations under the License.
+*
+**************************************************************************/
+
+#include <iostream>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "test_helper.hpp"
+#include "test_common.hpp"
+#include <gtest/gtest.h>
+
+extern std::vector<sycl::device*> devices;
+
+namespace {
+
+constexpr std::int64_t default_1d_lengths = 4;
+const std::vector<std::int64_t> default_3d_lengths{ 124, 5, 3 };
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+inline void set_and_get_lengths(sycl::queue& sycl_queue) {
+    /* Negative Testing */
+    {
+        oneapi::mkl::dft::descriptor<precision, domain> descriptor{ default_3d_lengths };
+        EXPECT_THROW(descriptor.set_value(oneapi::mkl::dft::config_param::LENGTHS, nullptr),
+                     oneapi::mkl::invalid_argument);
+    }
+
+    /* 1D */
+    {
+        oneapi::mkl::dft::descriptor<precision, domain> descriptor{ default_1d_lengths };
+
+        std::int64_t lengths_value{ 0 };
+        std::int64_t new_lengths{ 2345 };
+        std::int64_t dimensions_before_set{ 0 };
+        std::int64_t dimensions_after_set{ 0 };
+
+        descriptor.get_value(oneapi::mkl::dft::config_param::LENGTHS, &lengths_value);
+        descriptor.get_value(oneapi::mkl::dft::config_param::DIMENSION, &dimensions_before_set);
+        EXPECT_EQ(default_1d_lengths, lengths_value);
+        EXPECT_EQ(dimensions_before_set, 1);
+
+        descriptor.set_value(oneapi::mkl::dft::config_param::LENGTHS, new_lengths);
+        descriptor.get_value(oneapi::mkl::dft::config_param::LENGTHS, &lengths_value);
+        descriptor.get_value(oneapi::mkl::dft::config_param::DIMENSION, &dimensions_after_set);
+        EXPECT_EQ(new_lengths, lengths_value);
+        EXPECT_EQ(dimensions_before_set, dimensions_after_set);
+
+        commit_descriptor(descriptor, sycl_queue);
+    }
+
+    /* >= 2D */
+    {
+        const std::int64_t dimensions = 3;
+
+        oneapi::mkl::dft::descriptor<precision, domain> descriptor{ default_3d_lengths };
+
+        std::vector<std::int64_t> lengths_value(3);
+        std::vector<std::int64_t> new_lengths{ 1, 2, 7 };
+        std::int64_t dimensions_before_set{ 0 };
+        std::int64_t dimensions_after_set{ 0 };
+
+        descriptor.get_value(oneapi::mkl::dft::config_param::LENGTHS, lengths_value.data());
+        descriptor.get_value(oneapi::mkl::dft::config_param::DIMENSION, &dimensions_before_set);
+
+        EXPECT_EQ(default_3d_lengths, lengths_value);
+        EXPECT_EQ(dimensions, dimensions_before_set);
+
+        descriptor.set_value(oneapi::mkl::dft::config_param::LENGTHS, new_lengths.data());
+        descriptor.get_value(oneapi::mkl::dft::config_param::LENGTHS, lengths_value.data());
+        descriptor.get_value(oneapi::mkl::dft::config_param::DIMENSION, &dimensions_after_set);
+
+        EXPECT_EQ(new_lengths, lengths_value);
```

Comment:
We are testing if the opensource descriptor can set/get the Length but are we ensuring that is being communicated to the intel oneMKL descriptor
for instance setting LENGTHS is not allowed on GPU/CPU, and would resutl in a `ERR_BAD_OPERATION`
so either, 
1. we vet and remove the set length. (maybe vet for more such cases)
2. in the initial implimentation get_value ensure that the value stored in the config_value is the same as the one returned by the intel oneMKL descriptor.

Reply:
So you are saying that the compute tests passsing is indicative of a proper communitation to the backend desc?
and so what would happen if say I need to reset the strides of a real problem, and recommit before launching backward transform after running a forward transform ([e.g](https://github.com/intel-innersource/libraries.performance.math.mkl/blob/549fb74ecc8d528446fd57c22ac8a6c954b6578d/src/examples/dpcpp/dft/source/dp_real_2d_outofplace.cpp#L457)); if we can't set value on a commited descriptor then how do we impliment a real outofplace roundtrip with no padded data?

Reply:
1. Yes, I can't imagine a scenario where all tests pass without the backend getting the right configuration.
2. The case you are describing could work with 2 descriptors. This was our understanding from the [spec](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/dft/descriptor.html#commit): `All calls to the set_value function to change configuration parameters of a descriptor need to happen after the constructor call for the descriptor class and before a call to commit`. Maybe this is just confusing wording? Do we want to allow the user to `set_value` again as long it is committed again? Even if that is the case, the `LENGTHS` can still be writeable as we create a new backend descriptor when `commit` is called.

Reply:
1. Thank you, yeah that sounds fair,
2. Good point the spec does say that but as you can see in the examples we can keep the same descriptor and reuse it, for multiple cases. We should not recreate the discriptor unless its out of scope and the unique_ptr naturally destroy's it self. More on Lenghts (specific to the C Api)
`DFTI_DIMENSION and DFTI_LENGTHS do not have a default value. To set them, use the DftiCreateDescriptor function and not the DftiSetValue function.` ([ref](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/fourier-transform-functions/fft-functions/configuration-settings/dfti-dimension-dfti-lengths.html))

Reply:
We would prefer if we can update the open-source spec first if you need to make it closer to the closed-source MKL spec. I'm not seeing any real reason why we should change the current spec though. The descriptor likely needs to prepare again from scratch for any new commit, doesn't it? Also we're wondering if there could be complications if the user commits a second time with a different queue.